Output stress center

endrinar opened the pull request:
Tested with UnstructuredMeshTest.m

Commit:
Rename variable scalar_data to nodal_data

more descriptive

Commit:
Edit WriteMesh2VTK to output vectors

Commit:
Edit write2vtk_static to convert scalars to vectors

Commit:
Update UnstructuredMeshTest

Commit:
Fix bug in write2vtk_static - Fext should be R

Commit:
Update getStrain to output stress/strain at element center

b3gee commented:
- [x] Reviewed function(s): write2vtk, getStrain, WriteMesh2VTK
- [x] Ran test case(s): UnstructuredMeshTest with contour settings 'none', 'nodal', and 'center'
* Purpose of the code: To output the stress and strain at the element centroid
* Notes on implementation: 
   * Consider removing the second outputted .vtk file with the magnified displacement. If **u** is output as a vector in Paraview, we can now instead use the "Warp by vector" function in Paraview to visualize the deformation.
* Notes on legibility and style: 
   * This enhancement cleans up the outputted .vtk files nicely and makes them much more legible.
   * Please clarify the dependency between this branch and the **output-vectors** branch. It seems like this branch is directly working off **output-vectors**, yet the two are different branches with different pull requests.
   * The comments on the config files computation controls are misleading because it is labelled as nodal averaging. Consider clarifying the comments to better describe what these options do.
* Notes on maintainability: 
   * Please update the comments in MaterConfigFile to reflect the new stress output options.
* Suggested test cases: 
   * Please merge **develop** into this branch so that we can run the unit tests and make sure none of them are broken by this enhancement.

b3gee reviewed (changes requested):
See earlier comment for requested changes.

Commit:
Modify description of stress calculation variables

Changed variable naming to be more clear

Commit:
Merge branch 'develop' into output-stress-center

endrinar commented:
> * Notes on implementation:
>   
>   * Consider removing the second outputted .vtk file with the magnified displacement. If **u** is output as a vector in Paraview, we can now instead use the "Warp by vector" function in Paraview to visualize the deformation.

I agree. I have removed it now.

> * Notes on legibility and style:
>   
>   * This enhancement cleans up the outputted .vtk files nicely and makes them much more legible.
>   * Please clarify the dependency between this branch and the **output-vectors** branch. It seems like this branch is directly working off **output-vectors**, yet the two are different branches with different pull requests.

Yes, I can see how it is confusing. I completed the output-vectors branch first and created a pull request. Then I made this branch off of output-vectors with the sole purpose of computing the stress at the center of each element. I will need to look into how I can make it clear what needs to be reviewed for future pull requests of this sort. 

>   * The comments on the config files computation controls are misleading because it is labelled as nodal averaging. Consider clarifying the comments to better describe what these options do.

Agreed. I've updated the description and changed the variable naming throughout to make this more clear.

> * Notes on maintainability:
>   
>   * Please update the comments in MaterConfigFile to reflect the new stress output options.

Done.

> * Suggested test cases:
>   
>   * Please merge **develop** into this branch so that we can run the unit tests and make sure none of them are broken by this enhancement.

Done.

Thanks for the review.

Commit:
Modify description of stress/strain matrices in write2vtk_static

Commit:
Update Control.stress_calc in unit tests
